sdk/src/main/scala/com/horizen/validation/ConsensusValidator.scala

```diff
@@ -71,42 +71,69 @@ class ConsensusValidator extends HistoryBlockValidator with ScorexLogging {
       throw new SidechainBlockSlotInFutureException("Block had been generated in the future")
   }
 
+  /*
+      Visual schema for possible cases:
+      You can assume on different length of Consensus Epoch. For example, 7 or 9.
+      Block absolute slots: 3 - 12
+                                 |
+      Ommers slots:    [5    ,   7   ,   11]
+                        |        |        |
+      Subommers slots: [4]      [6]   [9  ,  10]
+                                       |
+      Subommers slots:                [8]
+      Ommer Container can include both the same consensus epoch ommers and/or previous consensus epoch ommers.
+      Inclusion, of ommers from 2 consensus epochs (or more) before is not valid, because it means,
+      that there is an empty epoch between Container and its parent, which is invalid case.
+      Most specific case when at some Ommers-level, there are both ommers from previous and current epoch.
+      In this case previous epoch ommers can lead to the current epoch Nonce different to the one known by our history.
+      It should be taken in consideration during Ommers VRF calculation,
+      so proper Nonce info should be used for such an Ommer with all sub-ommers recursively.
+   */
   private[horizen] def verifyOmmers(ommersContainer: OmmersContainer,
-                                     currentFullConsensusEpochInfo: FullConsensusEpochInfo,
-                                     previousFullConsensusEpochInfo: FullConsensusEpochInfo,
-                                     history: SidechainHistory): Unit = {
+                                    currentFullConsensusEpochInfo: FullConsensusEpochInfo,
+                                    previousFullConsensusEpochInfo: FullConsensusEpochInfo,
+                                    bestKnownParentId: ModifierId,
+                                    bestKnownParentInfo: SidechainBlockInfo,
+                                    history: SidechainHistory,
+                                    previousEpochOmmersInfoAccumulator: Seq[(VrfOutput, ConsensusSlotNumber)]
```

Comment:
I would prefer to have a Seq, because having an immutable data structure is more likely to use by passing it recursively to several "branches" that will further modify it, without modification of original one.